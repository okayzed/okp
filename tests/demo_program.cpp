
#include <iostream>
#include <tuple>

using namespace std;



class MyClass {
  public:
    string s;
    MyClass() {
        cout << "instantiated new class mc" << endl; }

    ~MyClass() {
        cout << "destructor for mc called" << ' ' <<  s << endl; }

    auto set_string(auto a) {
        s = a; }

    auto get_string() {
        return s; } };



auto foo() {
    cout << "test function" << endl;
    return make_tuple(1,  2); };

int main() {
    cout << "foobar" << endl;

    string s;
    cin >> s;
    cout << "READ" << ' ' <<  s << endl;

    MyClass mc;
    mc.set_string(s);

    auto structuredArgs_0 = foo();
    auto a = get<0>(structuredArgs_0);
    auto b = get<1>(structuredArgs_0);

    cout << a << ' ' <<  b << endl;

    if (true) {
        cout << "true" << endl; }

    if (not false) {
        cout << "false" << endl; }

    for (auto i = 0; i < 10; i++) {
      cout << "I IS" << ' ' <<  i << endl; }

    return 0; };


