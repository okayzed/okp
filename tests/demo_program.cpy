#include <iostream>
#include <tuple>

using namespace std

/*
   ignore
   multi line comments
   please
*/

class MyClass:
  public:
    string s;
    MyClass():
        print "instantiated new class mc"

    ~MyClass():
        print "destructor for mc called" s

    def set_string(auto a):
        s = a

    def get_string():
        return s


// should auto prepare a tuple when returning multiple
def foo():
    print "test function"
    return 1, 2

def main():
    print "foobar"

    string s
    read s // strip comments
    print "READ" s

    MyClass mc
    mc.set_string(s)

    a, b = foo() // std:tie when assigning from multi

    print a b

    if true:
        print "true"

    if not false:
        print "false"

    for i = 0; i < 10; i++:
      print "I IS" i

    return 0
