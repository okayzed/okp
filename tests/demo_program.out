foobar
READ data
instantiated new class mc
test function
1 2
true
false
I IS 0
I IS 1
I IS 2
I IS 3
I IS 4
I IS 5
I IS 6
I IS 7
I IS 8
I IS 9
destructor for mc called data
